webpack/scenes/Subscriptions/Manifest/CdnConfigurationForm.js

```diff
@@ -0,0 +1,166 @@
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+
+import {
+  ActionGroup,
+  Button,
+  Form,
+  FormGroup,
+  FormSelect,
+  FormSelectOption,
+  Spinner,
+  TextInput,
+} from '@patternfly/react-core';
+
+import { translate as __ } from 'foremanReact/common/I18n';
+
+import EditableTextInput from '../../../components/EditableTextInput';
+
+import {
+  selectUpdatingCdnConfiguration,
+} from '../../Organizations/OrganizationSelectors';
+
+import { updateCdnConfiguration } from '../../Organizations/OrganizationActions';
+import './CdnConfigurationForm.scss';
+
+const CdnConfigurationForm = (props) => {
+  const {
+    contentCredentials,
+    cdnConfiguration,
+  } = props;
+
+  const dispatch = useDispatch();
+
+  const [url, setUrl] = useState(cdnConfiguration.url);
+  const [username, setUsername] = useState(cdnConfiguration.username);
+  const [password, setPassword] = useState(null);
+  const [organizationLabel, setOrganizationLabel] =
+    useState(cdnConfiguration.upstream_organization_label);
+  const [sslCaCredentialId, setSslCaCredentialId] = useState(cdnConfiguration.ssl_ca_credential_id);
+  const updatingCdnConfiguration = useSelector(state => selectUpdatingCdnConfiguration(state));
+
+  const editPassword = (value) => {
+    if (value === null) {
+      setPassword('');
+    } else {
+      setPassword(value);
+    }
+  };
+
+  const hasPassword = (cdnConfiguration.password_exists && password === null)
+    || password?.length > 0;
+
+  const performUpdate = () => {
+    dispatch(updateCdnConfiguration({
+      url,
+      username,
+      password,
+      upstream_organization_label: organizationLabel,
+      ssl_ca_credential_id: sslCaCredentialId,
+    }));
+  };
+
+  const updateButtonContent = (
+    updatingCdnConfiguration ? (
+      <React.Fragment>
+        <Spinner size="md" />
+        {__('Updating')}
+      </React.Fragment>) :
+      __('Update')
+  );
+
+  return (
+    <div id="cdn-configuration">
+      <Form isHorizontal>
+        <FormGroup
+          label={__('URL')}
+          isRequired
+        >
+          <TextInput
+            isRequired
+            aria-label="cdn-url"
+            type="text"
+            value={url || ''}
+            onChange={value => setUrl(value)}
+          />
+        </FormGroup>
+        <FormGroup
+          label={__('Username')}
+        >
+          <TextInput
+            aria-label="cdn-username"
+            type="text"
+            value={username || ''}
+            onChange={value => setUsername(value)}
+          />
+        </FormGroup>
+        <FormGroup
+          label={__('Password')}
+        >
+          <EditableTextInput
+            attribute="cdn-password"
+            value={password}
+            isPassword
+            hasPassword={hasPassword}
+            onEdit={editPassword}
+          />
+        </FormGroup>
+        <FormGroup
+          label={__('Organization Label')}
```

Comment:
```suggestion
          label={__('Upstream Organization Label')}
```
Thoughts?  This would've saved me some confusion.

Reply:
Mostly -1 on this because I think you got the word upstream from the api parameter. that is named with upstream_ because hammer would have gotten confused when doing this: `hammer organization update-cdn-configuration --organization-id=X ...`

Confusion may be getting added to the mix here due to testing this locally against your own Katello.  A user would never do that and the documentation will always reflect an external source, so "upstream" is implied